lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -112,6 +115,7 @@ function NxComboboxRender<T extends string | number | DataItem<string | number,
         onChange(elToFocusMatch.displayName, elToFocusMatch);
       }
       setFocusableBtnIndex(null);
+      setHiddenBySelection(true);
```

Comment:
Should this line be here or in the `if (autoComplete...` condition above? My understanding is that we want to set `hiddenBySelection` if and only if we are firing `onChange` with its optional second argument.  In fact maybe a helper function should be written which combines those two actions, and all existing calls to `onChange(foo, bar)` replaced with that helper function.

Reply:
> Should this line be here or in the `if (autoComplete...` condition above? My understanding is that we want to set `hiddenBySelection` if and only if we are firing `onChange` with its optional second argument. In fact maybe a helper function should be written which combines those two actions, and all existing calls to `onChange(foo, bar)` replaced with that helper function.

Yep, it should have been, thanks for catching that. I did move it and created a helper function here: [f4ac895](https://github.com/sonatype/sonatype-react-shared-components/pull/876/commits/f4ac89526e30ad86244aa8f94f1b1b257b670e13). I added the third parameter and a condition inside of `handleSelection` because `onChange(foo, bar)` is also called in `adjustBtnFocus` where we don't want to handle `hiddenBySelection`.

Reply:
> Yep, it should have been, thanks for catching that. I did move it and created a helper function here: [f4ac895](https://github.com/sonatype/sonatype-react-shared-components/pull/876/commits/f4ac89526e30ad86244aa8f94f1b1b257b670e13). I added the third parameter and a condition inside of handleSelection because onChange(foo, bar) is also called in adjustBtnFocus where we don't want to handle hiddenBySelection.

Oh interesting, I hadn't thought about the arrow down case.  I think a cleaner way to do that would just be to use `onChange` directly in `adjustBtnFocus` and to avoid the conditional in `handleSelection`.  I also think `handleSelection` deserves a brief comment about what it does and when it should be used.

Reply:
> Oh interesting, I hadn't thought about the arrow down case. I think a cleaner way to do that would just be to use `onChange` directly in `adjustBtnFocus` and to avoid the conditional in `handleSelection`. I also think `handleSelection` deserves a brief comment about what it does and when it should be used.


I noticed a discrepancy in examples of  when showDropdown is true vs. when the dropdown should actually be visible. If the example is not set to auto complete, `showDropdown` remains `true` when you click outside of the dropdown without selecting anything (since no selection was made to change `hiddenBySelection(true)` ), but  is `false` when the `autoComplete` prop is true,  because the function `setHiddenBySelection` is contained inside the condition `if (autoComplete && focusableBtnindex !-- null)` . 

If this is to be consistent across all examples, we would need to have `setHiddenBySelection(true)` outside of the autoComplete condition, while needing to leave `onChange` inside based on your previous comment in this thread. IF that's the case, is it necessary to have an additional helper function that can only be called inside `handleDropdownBtnClick`?   ... this may also lead to needing to rename the state, since it would also be set to true when input loses focus

Reply:
Based on previous comment & discussion on Slack, conditional rendering implemented here: [5f0517b](https://github.com/sonatype/sonatype-react-shared-components/pull/876/commits/5f0517bc26bccd1922d2dfd4d4172be7833fabbe)

> > Oh interesting, I hadn't thought about the arrow down case. I think a cleaner way to do that would just be to use `onChange` directly in `adjustBtnFocus` and to avoid the conditional in `handleSelection`. I also think `handleSelection` deserves a brief comment about what it does and when it should be used.
> 

Update of `handleSelection` function here: [63002d8](https://github.com/sonatype/sonatype-react-shared-components/pull/876/commits/63002d8b32862f4b11bcc657fc3d629feedcf61e)